drone/exec.go

```diff
@@ -382,12 +384,46 @@ func exec(c *cli.Context) error {
 			Status: c.String("prev.build.status"),
 			Commit: c.String("prev.commit.sha"),
 		},
-		Job: &model.Job{
+	}
+
+	if len(c.StringSlice("matrix")) > 0 {
+		p := *payload
+		p.Job = &model.Job{
 			Environment: getMatrix(c),
-		},
+		}
+		return a.Run(&p, cancelc)
+	}
+
+	axes, err := yaml.ParseMatrix(file)
+	if err != nil {
+		return err
+	}
+
+	if len(axes) == 0 {
+		axes = append(axes, yaml.Axis{})
+	}
+
+	var jobs []*model.Job
+	count := 0
+	for _, axis := range axes {
+		jobs = append(jobs, &model.Job{
+			Number:      count,
+			Environment: axis,
+		})
+		count++
+	}
+
+	for _, job := range jobs {
+		fmt.Printf("Running Matrix job #%d\n", job.Number)
+		p := *payload
+		p.Job = job
+		p.Build.Number = p.Build.Number + job.Number
```

Comment:
this line should get removed. The build number is set using `DRONE_BUILD_NUMBER`  and this would override. All build meta-data is exposed via environment variables